common/dev/adc.c

```diff
@@ -98,28 +109,18 @@ static bool adc_read_mv(uint8_t sensor_num, uint32_t index, uint32_t channel, in
 	}
 	return false;
 }
-bool pal_adc_read(uint8_t sensor_num, int *reading)
+
+bool adc_sensor_read(uint8_t sensor_num, float *reading)
 {
-	uint8_t snrcfg_sensor_num = SnrNum_SnrCfg_map[sensor_num];
-	uint8_t chip = sensor_config[snrcfg_sensor_num].port / ADC_CHAN_NUM;
-	uint8_t number = sensor_config[snrcfg_sensor_num].port % ADC_CHAN_NUM;
+	snr_cfg *cfg = &sensor_config[sensor_config_index_map[sensor_num]];
+	uint8_t chip = cfg->port / ADC_CHAN_NUM;
+	uint8_t number = cfg->port % ADC_CHAN_NUM;
 	int val = 1;
 	int ret = adc_read_mv(sensor_num, chip, number, &val);
 	if (ret) {
-		if (sensor_num == SENSOR_NUM_VOL_BAT3V) {
-			gpio_set(A_P3V_BAT_SCALED_EN_R, GPIO_HIGH);
-			osDelay(1);
-			adc_read_mv(sensor_num, chip, number, &val);
-			val = val * sensor_config[snrcfg_sensor_num].arg0 /
-			      sensor_config[snrcfg_sensor_num].arg1;
-			gpio_set(A_P3V_BAT_SCALED_EN_R, GPIO_LOW);
-		} else {
-			val = val * sensor_config[snrcfg_sensor_num].arg0 /
-			      sensor_config[snrcfg_sensor_num].arg1;
-		}
-		*reading = (calculate_accurate_MBR(sensor_num, val) / 1000) & 0xFFFF;
-		sensor_config[snrcfg_sensor_num].cache = *reading;
-		sensor_config[snrcfg_sensor_num].cache_status = SNR_READ_ACUR_SUCCESS;
+		*reading = (float)val * cfg->arg0 / cfg->arg1 / 1000;
+		cfg->cache = *reading;
```

Comment:
check reading != NULL before use

Reply:
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

Reply:
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .